doc/tsm/tsi.md

```diff
@@ -35,19 +33,49 @@ These measurement blocks may duplicate measurement blocks created earlier by oth
 
 As TSI files grow and measurements are duplicated, query time merging becomes more costly so a larger compaction will be necessary to consolidate all series, measurement, and tag value blocks into one.
 
+When an index file or WAL file is compacted it is locked and the new index and WAL will begin on a new file. This is to allow snapshotting to take hard links on the files.
```

Comment:
Instead of locking the index and wal, what about rolling over the current wal into a new wal segments to writes that occur during the compaction are not blocked?  When the compaction completes, it would be a new index plus the remaining rolled over wal segments?